src/detect-bytetest.c

```diff
@@ -382,6 +442,9 @@ static DetectBytetestData *DetectBytetestParse(const char *optstr, char **value,
                 data->flags |= DETECT_BYTETEST_LITTLE;
             } else if (strcasecmp("dce", args[i]) == 0) {
                 data->flags |= DETECT_BYTETEST_DCE;
+            } else if (strncasecmp("bitmask", args[i], strlen("bitmask")) == 0) {
```

Comment:
why not a simple strcmp instead? Are we matching a partial string?

Reply:
Good question. This is not a substring so it could be just `strcasecmp` (like most, but not all, of the existing checks).

I'd prefer a case-insensitive match here to `strcasecmp` would work. The original code has a case-sensitive check for `relative` and a quick cross-check with the other `byte_` parsers shows inconsistent use of the three `str*cmp` functions.

What's your preference? Case-sensitive seems correct but case-insensitive would likely be ok and be detected easily with a rule parse error.

Reply:
@victorjulien preference?

Reply:
Lets use `strcmp` for new options. I don't want to break existing rules, so existing `strcasecmp` will have to stay as is.

Reply:
> why not a simple strcmp instead? Are we matching a partial string?

Sigh -- my previous answer was incorrect. Due to the PCRE limitation, the string being checked is of the form `bitmask <value>` so the `strncasecmp` is necessary.